### PR TITLE
CLOUDP-180579: Transform region config to be single object

### PR DESCRIPTION
**Pull request by wtrocki**
Based on the SDK migration I propose we squash region configs into a single object. 
Without that SDK usage is quite difficult which was proven in the CLI.

### CODE REVIEW

**Comment by wtrocki** [comment]
Draft because it needs fix in the examples.

**Comment by wtrocki** [comment]
This change requires team wide awareness. Creating micro proposal.

File: admin/model_region_config.go [hunk 1]
```diff
@@ -4,198 +4,370 @@ package admin
...
+// checks if the RegionConfig type satisfies the MappedNullable interface at compile time
+var _ MappedNullable = &RegionConfig{}
+
+// RegionConfig Cloud service provider on which MongoDB Cloud provisions the hosts.
 type RegionConfig struct {
-	AWSRegionConfig    *AWSRegionConfig
-	AzureRegionConfig  *AzureRegionConfig
-	GCPRegionConfig    *GCPRegionConfig
-	TenantRegionConfig *TenantRegionConfig
+	AnalyticsAutoScaling *AutoScalingV15        `json:"analyticsAutoScaling,omitempty"`
```
Comment (wtrocki) [inline admin/model_region_config.go:14]: Documentation for the merged object looks very good. It doesn't have any AWS, GCP specific comments as descriptions are missing from the fields that are merged.

File: admin/model_aws_region_config.go [hunk 2]
```diff
@@ -1,370 +0,0 @@
...
-// checks if the AWSRegionConfig type satisfies the MappedNullable interface at compile time
-var _ MappedNullable = &AWSRegionConfig{}
-
-// AWSRegionConfig Details that explain how MongoDB Cloud replicates data in one region on the specified MongoDB database.
-type AWSRegionConfig struct {
-	AnalyticsAutoScaling *AutoScalingV15        `json:"analyticsAutoScaling,omitempty"`
-	AnalyticsSpecs       *DedicatedHardwareSpec `json:"analyticsSpecs,omitempty"`
-	AutoScaling          *AutoScalingV15        `json:"autoScaling,omitempty"`
-	ReadOnlySpecs        *DedicatedHardwareSpec `json:"readOnlySpecs,omitempty"`
-	ElectableSpecs       *HardwareSpec          `json:"electableSpecs,omitempty"`
```
Comment (wtrocki) [inline admin/model_aws_region_config.go:18]: Those objects are common between parents.

**Review by swattyT — APPROVED** [review]
LGTM. Thanks a ton!

**Review by gssbzn — APPROVED** [review]

**Comment by wtrocki** [comment]
Will follow up with release.

**Pull request merged**